src/lib.rs

```diff
@@ -306,13 +306,25 @@ impl KVMIntrospectable for KVMi {
         Ok(vcpu_count)
     }
 
-    fn get_registers(&self, vcpu: u16) -> Result<(kvm_regs, kvm_sregs, kvm_msrs), Error> {
+    fn get_registers(&self, vcpu: u16) -> Result<(kvm_regs, kvm_sregs, KvmMsr), Error> {
         let mut regs: kvm_regs = unsafe { mem::MaybeUninit::<kvm_regs>::zeroed().assume_init() };
         let mut sregs: kvm_sregs = unsafe { mem::MaybeUninit::<kvm_sregs>::zeroed().assume_init() };
-        let mut msrs: kvm_msrs = unsafe { mem::MaybeUninit::<kvm_msrs>::zeroed().assume_init() };
+        let mut msrs: KvmMsr = unsafe { mem::MaybeUninit::<KvmMsr>::zeroed().assume_init() };
         let mut mode: c_uint = 0;
+        msrs.msrs.nmsrs = 6;
+        msrs.entries[0].index = KVMiMsr::SysenterCs as u32;
+        msrs.entries[1].index = KVMiMsr::SysenterEsp as u32;
```

Comment:
nitpick: There are currently 3 entities you are working with here: `kvm_msrs`, `KvmMsr` and `KVMiMsr`. This is really confusing to read. I'd suggest something like `KvmMsrIndices` instead of `KVMiMsr`.